### PR TITLE
Add support for volatile stores

### PR DESCRIPTION
**Pull request by phate**
This PR does the following:

    Add StoreOperation and StoreNode class for volatile load operations.
    Adds conversion support for LLVM -> RVSDG
    Adds conversion support for RVSDG -> LLVM

### CODE REVIEW

File: jlm/llvm/backend/jlm2llvm/instruction.cpp [hunk 1]
```diff
@@ -324,18 +324,45 @@ convert(
...
+      storeOperation->GetAlignment(),
+      builder,
+      ctx);
+  return nullptr;
+}
 
-  auto i = builder.CreateStore(ctx.value(args[1]), ctx.value(args[0]));
-  i->setAlignment(::llvm::Align(store->GetAlignment()));
+static ::llvm::Value *
+convert(
```
Comment (haved) [inline jlm/llvm/backend/jlm2llvm/instruction.cpp:359]: I assume there is a historic reason why the above function is called `convert_store` and takes an untyped `simple_op`, while this one doesn't?

File: jlm/llvm/ir/operators/StoreVolatile.hpp [hunk 2]
```diff
@@ -0,0 +1,219 @@
...
+
+public:
+  [[nodiscard]] const StoreVolatileOperation &
+  GetOperation() const noexcept
+  {
+    return *util::AssertedCast<const StoreVolatileOperation>(&operation());
+  }
+
+  [[nodiscard]] size_t
+  NumMemoryStates() const noexcept
```
Comment (haved) [inline jlm/llvm/ir/operators/StoreVolatile.hpp:134]: Are there no helper functions for `GetMemoryStateInput(index)` and `GetMemoryStateOutput(index)`?

File: jlm/llvm/backend/jlm2llvm/instruction.cpp (hunk 1 above)
Comment (phate) [reply jlm/llvm/backend/jlm2llvm/instruction.cpp:359]: Yes, some cleanup and refactoring is necessary in this pass. However, that is for another PR.

File: jlm/llvm/ir/operators/StoreVolatile.hpp (hunk 2 above)
Comment (phate) [reply jlm/llvm/ir/operators/StoreVolatile.hpp:134]: Meh, I forgot them. It was also not needed when writing the tests. I will add them once they are required. I want to get this merged and move on.

**Pull request merged**